File: app/src/main/java/com/example/contacts/database/converter/DateConverter.java
package com.example.contacts.database.converter;

import androidx.room.TypeConverter;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class DateConverter {

    @TypeConverter
    public static Long dateToTimestamp(String date) {
        if (date == null) {
            return null;
        } else {
            try {
                Date localDate = new SimpleDateFormat("dd.mm.yyyy").parse(date);
                return localDate.getTime();
            } catch (ParseException e) {
                return null;
            }
        }
    }

    @TypeConverter
    public static String dateToString(Long date){
        if (date == null)
            return null;
        else{
            Date localDate = new Date(date);
            DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
            return dateFormat.format(localDate);
        }
    }
}

File: app/src/main/java/com/example/contacts/database/converter/DateConverter.kt
package com.example.contacts.database.converter

import androidx.room.TypeConverter
import java.text.DateFormat
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateConverter {
    @JvmStatic
    @TypeConverter
    fun dateToTimestamp(date: String?): Long? {
        return if (date == null) {
            null
        } else {
            try {
                val localDate = SimpleDateFormat("dd.mm.yyyy").parse(date)
                localDate!!.time
            } catch (e: ParseException) {
                null
            }
        }
    }

    @JvmStatic
    @TypeConverter
    fun dateToString(date: Long?): String? {
        return if (date == null) null else {
            val localDate = Date(date)
            val dateFormat: DateFormat = SimpleDateFormat("dd.MM.yyyy")
            dateFormat.format(localDate)
        }
    }
}